dht_options.go

```diff
@@ -315,3 +318,15 @@ func customProtocols(protos ...protocol.ID) Option {
 		return nil
 	}
 }
+
+// V1CompatibleMode sets the DHT to operate in V1 compatible mode. In this mode,
+// the DHT node will act like a V1 DHT node (use the V1 protocol names) but will
+// use the V2 query and routing table logic.
+//
+// This option is perma-unstable and may be removed in the future.
+func V1CompatibleMode() Option {
```

Comment:
probably want a bit of thought on what interface gets exposed here, since it may stick around for a while.

Reply:
I'm not sure if we can go with anything simpler than "enable/disable".